File: units_java/src/test/java/com/units/tests/constants/TestUranusProperties.java
package com.units.tests.constants;

import static com.units.constants.PhysicalConstants.gravitationalConstant;
import static com.units.constants.UranusProperties.*;

import org.junit.Test;

import com.units.acceleration.MetersPerSquareSecond;
import com.units.mass.Kilograms;
import com.units.speed.MetersPerSecond;

public class TestUranusProperties extends PlanetaryConstantsTestSuite {

	public TestUranusProperties() {
		super(2.4); // %
	}
	
	@Test
	public void testSurfaceGravity() {
		MetersPerSquareSecond calculatedAcceleration =
				gravitationalConstant
				.multiply(uranusMass)
				.divide(uranusEquatorialRadius)
				.divide(uranusEquatorialRadius)
				.castTo(MetersPerSquareSecond.class);
		checkAcceleration(uranusSurfaceGravity, calculatedAcceleration);	
	}
	
	@Test
	public void testMass() {
		Kilograms calculatedMass =
				uranusMeanDensity
				.multiply(uranusEquatorialRadius.pow(3))
				.castTo(Kilograms.class)
				.multiplyByScalar(Math.PI * 4 / 3);
		checkMass(uranusMass, calculatedMass);
	}
	
	@Test
	public void testEscapeVelocity() {
		MetersPerSecond calculatedEscapeVelocity =
				gravitationalConstant
				.multiply(uranusMass)
				.divide(uranusEquatorialRadius)
				.sqrt()
				.castTo(MetersPerSecond.class)
				.multiplyByScalar(Math.sqrt(2));
		checkVelocity(uranusEscapeVelocity, calculatedEscapeVelocity);
	}

}

File: units_cpp/src/unitsTest/cpp/constants/test_uranus_properties.cc
/*
 * test_uranus_properties.cc
 *
 *  Created on: Dec 27, 2017
 *      Author: sagis
 */

#include <units/constants/uranus_properties.h>
#include <units/constants/physical_constants.h>
#include <units/units_multiply.h>
#include <units/units_ratio.h>
#include <units/units_pow.h>
#include <units/units_sqrt.h>
#include <units/planetary_constants_test_suite.h>

using namespace units::uranus_properties;
using namespace units;
using units::physical_constants::gravitationalConstant;

class TestUranusProperties : public PlanetaryConstantsTestSuite
{
protected:
	TestUranusProperties()
	: PlanetaryConstantsTestSuite{2.4} // %
	{
	}
};

TEST_F(TestUranusProperties, test_surface_gravity)
{
	auto calculatedSurfaceGravity =
			units_ratio<units::acceleration::MetersPerSquareSecond>(
					auto_units_multiply(gravitationalConstant, uranusMass),
					auto_units_multiply(uranusEquatorialRadius, uranusEquatorialRadius));
	checkEqualValue(uranusSurfaceGravity, calculatedSurfaceGravity);
}

TEST_F(TestUranusProperties, test_mass)
{
	auto calculatedMass = (M_PI * 4 / 3) *
			units_multiply<units::mass::Kilograms>(
					uranusMeanDensity ,units_pow<3>(uranusEquatorialRadius));
	checkEqualValue(uranusMass, calculatedMass);
}

TEST_F(TestUranusProperties, test_escape_velocity)
{
	auto calculatedEscapeVelocity = units_sqrt<units::speed::MetersPerSecond> (
			2 * auto_units_ratio(
					auto_units_multiply(gravitationalConstant, uranusMass),
					uranusEquatorialRadius
				)
			);
	checkEqualValue(uranusEscapeVelocity, calculatedEscapeVelocity);
}
